File: EPZEU_PR/property-register/pr-web/pr-web-public/src/main/webapp/app/EPZEU.PR.Applications/Models/Applications/ApplicationForCertificateForProperty.ts
import { ApplicationFormBase } from 'EPZEU.PR.ApplicationBase';
import { TypeSystem } from 'Cnsys.Core';
import { moduleContext } from '../../ModuleContext';
import { observable } from "mobx";
import { RequestedProperty } from "../Sections/RequestedProperty";
import { Owners } from "../Sections/Owners";
import { WayOfProvision } from "../Sections/WayOfProvision";
import { ContactData } from "../Sections/ContactData";
import { Individual } from "../Individual";
import {InitialApplicationData} from "../Sections/InitialApplicationData";
import {GdprAgreement} from "../Sections/GdprAgreement";

@TypeSystem.typeDecorator('ApplicationForCertificateForProperty', moduleContext.moduleName)
export class ApplicationForCertificateForProperty extends ApplicationFormBase {
  @observable private _applicantData: Individual = null;
  @observable private _requestedProperty: RequestedProperty = null;
  @observable private _currentOwners: Owners = null;
  @observable private _previousOwners: Owners = null;
  @observable private _wayOfProvision: WayOfProvision = null;
  @observable private _contactData: ContactData = null;
  @observable private _initialApplicationData: InitialApplicationData = null;
  @observable private _gdprAgreement: GdprAgreement = null;

  constructor(obj?: any) {
    super(obj);

    this.copyFrom(obj);
  }

  public get applicantData(): Individual {
    return this._applicantData;
  }

  @TypeSystem.propertyDecorator(Individual)
  public set applicantData(value: Individual) {
    this._applicantData = value;
  }

  public get requestedProperty(): RequestedProperty {
    return this._requestedProperty;
  }

  @TypeSystem.propertyDecorator(RequestedProperty)
  public set requestedProperty(value: RequestedProperty) {
    this._requestedProperty = value;
  }

  public get currentOwners(): Owners {
    return this._currentOwners;
  }

  @TypeSystem.propertyDecorator(Owners)
  public set currentOwners(value: Owners) {
    this._currentOwners = value;
  }

  public get previousOwners(): Owners {
    return this._previousOwners;
  }

  @TypeSystem.propertyDecorator(Owners)
  public set previousOwners(value: Owners) {
    this._previousOwners = value;
  }

  public get wayOfProvision(): WayOfProvision {
    return this._wayOfProvision;
  }

  @TypeSystem.propertyDecorator(WayOfProvision)
  public set wayOfProvision(value: WayOfProvision) {
    this._wayOfProvision = value;
  }

  public get contactData(): ContactData {
    return this._contactData;
  }

  @TypeSystem.propertyDecorator(ContactData)
  public set contactData(value: ContactData) {
    this._contactData = value;
  }

  public get initialApplicationData(): InitialApplicationData {
    return this._initialApplicationData;
  }

  @TypeSystem.propertyDecorator(InitialApplicationData)
  public set initialApplicationData(value: InitialApplicationData) {
    this._initialApplicationData = value;
  }

  get gdprAgreement(): GdprAgreement {
    return this._gdprAgreement;
  }
  @TypeSystem.propertyDecorator(GdprAgreement)
  set gdprAgreement(value: GdprAgreement) {
    this._gdprAgreement = value;
  }
}

File: EPZEU_PR/property-register/pr-application/src/main/java/bg/registryagency/epzeu/pr/application/ApplicationForCertificateForProperty.java
package bg.registryagency.epzeu.pr.application;

import bg.registryagency.epzeu.pr.application.segment.*;
import bg.registryagency.epzeu.pr.application.segment.mapper.IdentityDtoMapper;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.xml.bind.annotation.*;
import java.util.List;


/**
 * Заявление за издаване на удостоверение за лице
 * 
 * <p>Java class for ApplicationForIssuingOfCertificateForPropertyFromPropertyRegister complex type.
 * 
 * <p>The following schema fragment specifies the expected content contained within this class.
 * 
 * <pre>
 * &lt;complexType name="ApplicationForIssuingOfCertificateForPropertyFromPropertyRegister"&gt;
 *   &lt;complexContent&gt;
 *     &lt;restriction base="{http://www.w3.org/2001/XMLSchema}anyType"&gt;
 *       &lt;sequence&gt;
 *         &lt;element name="ApplicationType" type="{http://ereg.egov.bg/value/0008-000144}ApplicationType"/&gt;
 *         &lt;element name="InitialApplicationData" type="{InitialApplicationData}InitialApplicationData" minOccurs="0"/&gt;
 *         &lt;element name="ApplicantData" type="{Individual}Individual"/&gt;
 *         &lt;element name="RequestedProperty" type="{Property}Property"/&gt;
 *         &lt;element name="CurrentOwners" type="{Owners}Owners"/&gt;
 *         &lt;element name="PreviousOwners" type="{Owners}Owners"/&gt;
 *         &lt;element name="WayOfProvision" type="{WayOfProvision}WayOfProvision"/&gt;
 *         &lt;element name="ContactData" type="{ContactData}ContactData"/&gt;
 *         &lt;element name="AttachedDocuments" type="{AttachedDocuments}AttachedDocuments" minOccurs="0"/&gt;
 *       &lt;/sequence&gt;
 *     &lt;/restriction&gt;
 *   &lt;/complexContent&gt;
 * &lt;/complexType&gt;
 * </pre>
 * 
 * 
 */
@Setter
@NoArgsConstructor
@XmlRootElement(name = "ApplicationForIssuingOfCertificateForPropertyFromPropertyRegister")
@XmlType(name = "ApplicationForIssuingOfCertificateForPropertyFromPropertyRegister", propOrder = {
    "applicationContentType",
    "initialApplicationData",
    "applicantData",
    "requestedProperty",
    "currentOwners",
    "previousOwners",
    "wayOfProvision",
    "contactData",
    "attachedDocuments",
    "signature",
    "gdprAgreement"
})
public class ApplicationForCertificateForProperty extends BaseApplicationForCertificate {

    protected Property requestedProperty;
    protected Owners currentOwners;
    protected Owners previousOwners;

    @XmlElement(name = "RequestedProperty", required = true)
    public Property getRequestedProperty() {
        return requestedProperty;
    }
    @XmlElement(name = "CurrentOwners", required = true)
    public Owners getCurrentOwners() {
        return currentOwners;
    }
    @XmlElement(name = "PreviousOwners", required = true)
    public Owners getPreviousOwners() {
        return previousOwners;
    }

    public ApplicationForCertificateForProperty(Individual applicantData) {
        this.applicantData = applicantData;
    }

    @Override
    public String getApplicantName() {
        return applicantData.getName().toString();
    }

    @Override
    public String getIdentity() {
        return IdentityDtoMapper.asDto(applicantData.getIdentity());
    }
}
